internal/controller/component_build_controller_service_account.go

```diff
@@ -0,0 +1,523 @@
+	pullPipelinePath := getPipelineRunDefinitionFilePath(component, true)
+
+	pushPipelineDefinitionFound := true
+	var pushPipelineContent []byte
+	pushPipelineContent, err = gitClient.DownloadFileContent(repoUrl, baseBranch, pushPipelinePath)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		pushPipelineDefinitionFound = false
+	}
+	pullPipelineDefinitionFound := true
+	var pullPipelineContent []byte
+	pullPipelineContent, err = gitClient.DownloadFileContent(repoUrl, baseBranch, pullPipelinePath)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		pullPipelineDefinitionFound = false
+	}
+	if !pushPipelineDefinitionFound && !pullPipelineDefinitionFound {
+		// No pipeline definitions found
+		log.Info("No pipeline definitions found to migrate")
+		return nil
+	}
+
+	pipelineRunServiceAccountName := getBuildPipelineServiceAccountName(component)
+
+	if pushPipelineDefinitionFound {
+		pushPipelineContent, err = addServiceAccountToPipelineRun(pushPipelineContent, pipelineRunServiceAccountName)
+		if err != nil {
+			log.Error(err, "failed to add Service Account to push pipeline definition")
+			return err
+		}
+	}
+	if pullPipelineDefinitionFound {
+		pullPipelineContent, err = addServiceAccountToPipelineRun(pullPipelineContent, pipelineRunServiceAccountName)
+		if err != nil {
+			log.Error(err, "failed to add Service Account to pull pipeline definition")
+			return err
+		}
+	}
+
+	mrData := &gp.MergeRequestData{
+		CommitMessage:  "Konflux build pipeline service account migration for " + component.Name,
+		SignedOff:      true,
+		BranchName:     fmt.Sprintf("%s%s", "konflux-sa-migration-", component.Name),
+		BaseBranchName: baseBranch,
+		Title:          "Konflux build pipeline service account migration",
+		Text:           saMigrationMergeRequestDescription,
+		AuthorName:     "konflux",
+		AuthorEmail:    "konflux@no-reply.konflux-ci.dev",
+		Files:          []gp.RepositoryFile{},
+	}
+	if pushPipelineDefinitionFound {
+		mrData.Files = append(mrData.Files, gp.RepositoryFile{FullPath: pushPipelinePath, Content: pushPipelineContent})
+	}
+	if pullPipelineDefinitionFound {
+		mrData.Files = append(mrData.Files, gp.RepositoryFile{FullPath: pullPipelinePath, Content: pullPipelineContent})
+	}
+
+	var webUrl string
+	webUrl, err = gitClient.EnsurePaCMergeRequest(repoUrl, mrData)
+	if err != nil {
+		log.Error(err, "failed to create migration PR")
+		return err
+	}
+
+	log.Info("Migration PR: " + webUrl)
+	return nil
+}
+
+// addServiceAccountToPipelineRun specifies given service account to use in the pipeline.
+func addServiceAccountToPipelineRun(pipelineRunBytes []byte, serviceAccountName string) ([]byte, error) {
+	buildPipelineData := &tektonapi.PipelineRun{}
+	if err := yaml.Unmarshal(pipelineRunBytes, buildPipelineData); err != nil {
+		return nil, err
+	}
+
+	buildPipelineData.Spec.TaskRunTemplate.ServiceAccountName = serviceAccountName
```

Comment:
you should 1st check if it isn't there already

and return additional boolean value indicating that

Reply:
Yeah, makes sense, than you, fixed.